Docs: Improve Nsight Systems Section (#3102)
ECP-WarpX/WarpX: Docs/source/developers/profiling.rst

    - You can directly click on the timeline to see which actual MPI call is being perform. (Note that the colorbar can be misleading.)

.. _developers-profiling-nsight-systems:

> .. _running-cpp-perlmutter-A100-GPUs:

Reviewer: This section is used somewhere else already (perlmutter docs: target of the link we use below)
```suggestion
```

.. code-block:: bash

> 

Reviewer: ```suggestion

   # GPU-aware MPI
   export MPICH_GPU_SUPPORT_ENABLED=1
   # 1 OpenMP thread
   export OMP_NUM_THREADS=1
   export TMPDIR="$PWD/tmp"
   rm -rf ${TMPDIR} profiling*
   mkdir -p ${TMPDIR}
   # 2021.5.1 (broken: lacks most trace info)
   #NSYS="/global/common/software/nersc/pm-2021q4/easybuild/software/Nsight-Systems/2021.5.1/bin/nsys"
   # 2021.4.1 (working)
   NSYS="/opt/nvidia/hpc_sdk/Linux_x86_64/21.11/compilers/bin/nsys"
   # record
```

.. code-block:: bash

>    #!/bin/bash -l

Reviewer: Hi @Thierry992 , please do not copy paste the header but use the `:ref:` feature of rst (restful text) that we discussed.
- https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html
- ref to https://github.com/ECP-WarpX/WarpX/blame/development/Docs/source/install/hpc/perlmutter.rst section `running-cpp-perlmutter-A100-GPUs`
```
:ref:`link text here <running-cpp-perlmutter-A100-GPUs>`
```

The reason for that:
- we do not need to use a header, you can do the same in an interactive job
- if something changes on Perlmutter, we only want to update one location in the docs, otherwise we will get lost in maintenance work if we duplicate it all over the place

Reviewer: Hi @Thierry992, can you please update the pull request accordingly? :-)

   export MPICH_GPU_SUPPORT_ENABLED=1
   # 1 OpenMP thread

>    export OMP_NUM_THREADS=1

Reviewer: ```suggestion
   export OMP_NUM_THREADS=1

```

   export TMPDIR="$PWD/tmp"
   rm -rf ${TMPDIR} profiling*

>    mkdir -p ${TMPDIR}

Reviewer: ```suggestion
   mkdir -p ${TMPDIR}

```

   # 2021.5.1 (broken: lacks most trace info)
   #NSYS="/global/common/software/nersc/pm-2021q4/easybuild/software/Nsight-Systems/2021.5.1/bin/nsys"
   # 2021.4.1 (working)

>    NSYS="/opt/nvidia/hpc_sdk/Linux_x86_64/21.11/compilers/bin/nsys"

Reviewer: ```suggestion
   NSYS="/opt/nvidia/hpc_sdk/Linux_x86_64/21.11/compilers/bin/nsys"

```

         inputs_3d                           \
           warpx.numprocs=1 1 4 amr.n_cell=512 512 2048 max_step=10

> You can find the header of the file :ref:`here <running-cpp-perlmutter-A100-GPUs>`.
> 

Reviewer: ```suggestion
```

   This version does not record all trace information.
   You need to use the one directly shipped with the NVHPC base system, version 2021.4.1, located in ``/opt/nvidia/hpc_sdk/Linux_x86_64/21.11/compilers/bin/nsys``.

.. note::

>     If everything went well, the user will obtain as many output files named "profiling_<number>.nsys-rep" as active GPU's which will allow the user to       analyze each of those GPU's perforamces with Nsight system.

Reviewer: Great idea to add what is expected!

Here are small suggestions: make the file name `verbatim` and remove some spaces.
```suggestion
    If everything went well, the user will obtain as many output files named ``profiling_<number>.nsys-rep`` as active MPI ranks, which will allow the user to analyze each MPI rank's performance trace with the Nsight System graphical user interface (GUI).
    In WarpX, every MPI rank is associated with one GPU.
```

   The Nsight-Compute (``nsys``) version installed on Summit does not record details of GPU kernels.
   This is reported to Nvidia and OLCF.

> 
> 
> Details

Reviewer: Please undo the unrelated change:
```suggestion
Details
```
